File: src/frontend/src/app/core/services/avaliador.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Avaliador, CreateAvaliador, UpdateAvaliador } from '../models';

@Injectable({
  providedIn: 'root'
})
export class AvaliadorService {
  private endpoint = '/avaliadores';

  constructor(private api: ApiService) {}

  getAll() {
    return this.api.get<Avaliador[]>(this.endpoint);
  }

  getById(id: number) {
    return this.api.get<Avaliador>(`${this.endpoint}/${id}`);
  }

  getByProcessoId(processoId: number) {
    return this.api.get<Avaliador[]>(`${this.endpoint}/processo/${processoId}`);
  }

  create(data: CreateAvaliador) {
    return this.api.post<Avaliador>(this.endpoint, data);
  }

  update(id: number, data: UpdateAvaliador) {
    return this.api.put<Avaliador>(`${this.endpoint}/${id}`, data);
  }

  delete(id: number) {
    return this.api.delete<void>(`${this.endpoint}/${id}`);
  }
}

File: src/frontend/ProcessoSelecao.Blazor/Services/AvaliadorService.cs
using ProcessoSelecao.Blazor.Models;

namespace ProcessoSelecao.Blazor.Services;

public class AvaliadorService
{
    private readonly ApiService _api;
    private const string Endpoint = "avaliadores";

    public AvaliadorService(ApiService api)
    {
        _api = api;
    }

    public async Task<List<Avaliador>> GetAllAsync()
        => await _api.GetAsync<List<Avaliador>>(Endpoint) ?? new();

    public async Task<Avaliador?> GetByIdAsync(int id)
        => await _api.GetAsync<Avaliador>($"{Endpoint}/{id}");

    public async Task<List<Avaliador>> GetByProcessoIdAsync(int processoId)
        => await _api.GetAsync<List<Avaliador>>($"{Endpoint}/processo/{processoId}") ?? new();

    public async Task<Avaliador?> CreateAsync(CreateAvaliador dto)
        => await _api.PostAsync<Avaliador>(Endpoint, dto);

    public async Task<Avaliador?> UpdateAsync(int id, UpdateAvaliador dto)
        => await _api.PutAsync<Avaliador>($"{Endpoint}/{id}", dto);

    public async Task DeleteAsync(int id)
        => await _api.DeleteAsync($"{Endpoint}/{id}");
}
